R/summarise_scores.R

```diff
@@ -58,93 +55,53 @@
 #' # round digits
 #' summarise_scores(scores,by = "model") %>%
 #'   summarise_scores(fun = signif, digits = 2)
-#'
-#' # get quantiles of scores
-#' # make sure to aggregate over interval ranges first
-#' summarise_scores(scores,
-#'   by = "model", fun = quantile,
-#'   probs = c(0.25, 0.5, 0.75)
-#' )
-#'
-#' # get ranges
-#' # summarise_scores(scores, by = "range")
 #' @export
+#' @importFrom checkmate assert_subset assert_function test_subset
 #' @keywords scoring
 
 summarise_scores <- function(scores,
-                             by = NULL,
+                             by = "model",
                              across = NULL,
                              fun = mean,
                              ...) {
-  if (!is.null(across) && !is.null(by)) {
-    stop("You cannot specify both 'across' and 'by'. Please choose one.")
-  }
-
+  # input checking ------------------------------------------------------------
+  # Check the score names attribute exists
   score_names <- attr(scores, "score_names")
   if (is.null(score_names)) {
-    stop("`scores` needs to have an attribute `score_names` with the names of
-         the metrics that were used for scoring.")
+    stop("`scores` needs to have an attribute `score_names` with ",
+         "the names of the metrics that were used for scoring.")
+  }
+
+  if (!test_subset(score_names, names(scores))) {
```

Comment:
we may want to make this into a little internal function as I can see it be reused. We also may want to help the user know howto update this if they need to. This is all another issue.